PowerShellGet/PSModule.psm1

```diff
@@ -4906,6 +4906,16 @@ function Test-ScriptFileInfo
             return
         }
 
+        if ($PSScriptInfo.Version -match '-')
+        {
+            # Version contains PreRelease string.  Break apart and validate separately.
+            $indexOfFirstHyphen = $PSScriptInfo.Version.IndexOf('-')
+            $versionSubstring = $PSScriptInfo.Version.Substring(0, $indexOfFirstHyphen)
+            $preReleaseSubstring = $PSScriptInfo.Version.Substring($indexOfFirstHyphen+1)
+
+            $preReleaseIsValid = Validate-PreReleaseString -Version $versionSubstring -PreRelease $preReleaseSubstring -InfoObject $PSScriptInfo

```

Comment:
Take a look at the usage of Validate-VersionParameters function. You also need to ensure that execution doesn't continue when -ErrorAction SilentlyContinue is specified.

---
In reply to: [126788397](https://github.com/rebro-msft/PowerShellGet/pull/1#discussion_r126788397) [](ancestors = 126788397)